web/src/components/molecules/Content/Table/DropdownRender.tsx

```diff
@@ -0,0 +1,295 @@
+import styled from "@emotion/styled";
+import moment from "moment";
+import { useRef, useEffect } from "react";
+import { useSearchParams } from "react-router-dom";
+
+import Button from "@reearth-cms/components/atoms/Button";
+import DatePicker, { DatePickerProps } from "@reearth-cms/components/atoms/DatePicker";
+import Form from "@reearth-cms/components/atoms/Form";
+import Input from "@reearth-cms/components/atoms/Input";
+import InputNumber from "@reearth-cms/components/atoms/InputNumber";
+import Select from "@reearth-cms/components/atoms/Select";
+import Space from "@reearth-cms/components/atoms/Space";
+import {
+  DefaultFilterValueType,
+  Operator,
+  DropdownFilterType,
+} from "@reearth-cms/components/molecules/Content/Table/types";
+import {
+  BasicOperator,
+  BoolOperator,
+  NullableOperator,
+  NumberOperator,
+  TimeOperator,
+  StringOperator,
+  SortDirection,
+} from "@reearth-cms/gql/graphql-client-api";
+import { useT } from "@reearth-cms/i18n";
+
+type Props = {
+  filter: DropdownFilterType;
+  close: () => void;
+  defaultValue?: DefaultFilterValueType;
+  open: boolean;
+  isFilter: boolean;
+  index: number;
+};
+
+const DropdownRender: React.FC<Props> = ({
+  filter,
+  close,
+  defaultValue,
+  open,
+  isFilter,
+  index,
+}) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const t = useT();
+  const [form] = Form.useForm();
+
+  useEffect(() => {
+    if (open && !defaultValue) {
+      form.resetFields();
+    }
+  }, [open, form, defaultValue]);
+
+  const options: {
+    operatorType: string;
+    value: Operator | SortDirection;
+    label: string;
+  }[] = [];
+
+  if (isFilter) {
```

Comment:
Wrap this inside a use memo with the if and switch statements
```suggestion
  const options = useMemo<{
    operatorType: string;
    value: Operator | SortDirection;
    label: string;
  }>(() => {
    const result = [];
   
    if (isFilter) {